### DISCUSSION TITLE
How to access style icons via std-widgets.slint

### DISCUSSION THREAD

**Discussion by develcooking** [slint-ui/slint#5234] [Q&A] [answered]
I'm currently working on an application that needs a scrollable combobox, so I wrote a custom widget with a bit of inspiration from @ogoffart in https://github.com/slint-ui/slint/discussions/2657. But how can I use the stylized icons in my custom widget?

I could just download them and use them on my own, but if I switch the style/theme from Fluent to Cosmic, the icons would look off.

Internally, they are already defined in, for example, Fluent style as an additional palette: https://github.com/slint-ui/slint/blob/28c6b8d0ba995eeb3ea053af50b76a7c89e6e97f/internal/compiler/widgets/fluent-base/styling.slint

So, how do I import them into my widget? Correct me if I'm wrong, but there isn't an easy solution, or is there?

**Comment by ghost** [comment]
Hi, 
with the icons is a difficult to topic. Right now only the `cosmic` and `material` style are using the right icons, because both icon sets are published under a free license. The apple icons are difficult to use because of the license and the icons for fluent should be free, but there are some restrictions.  So as backup we are using right now the `material` icons also for  `cupertino` and `fluent` as fallback. But in the end we will need more suitable icons for the styles. So in the moment we have only very small subsets of the icons, the icons we are needed for internal development. An other point is, if we would provide a full set of all icons, that would be a lot of svg files. So I'm not sure it make sense to deliver icons alongside with Slint. What I thought about is e.g. for material and cosmic to fork the icon repositories and create e.g. a crate from it that makes it easy to include the icons.

For my custom widget set I provide a subset of the `material` icons that can be used from it https://codeberg.org/flovansl/co_sl/src/branch/main/widgets/coop/ui/styling/icons.slint. Maybe if you need some inspiration.

**Reply by develcooking** [reply]
Well, that's sad to hear. Do you know about a different implementation to create a custom widget that fits into the existing styles? Even if it only would work for `material` and `cosmic`?

I don't want to use a different widget lib that wouldn't fit into the style of the OS it's running on.

**Reply by ghost** [reply, marked as answer]
I'm confident we will find a proper way for the icons in the future. Yes you can check the `Palette` global from `std-widgets` https://releases.slint.dev/1.6.0/docs/slint/src/language/builtins/globals.html#palette. It provides you different widget/control brushes for the current selected style.  We will make more styling resources public soon like font settings.
